### ISSUE TITLE
Suppress "All plugins found for ios are Swift Packages, but your project still has CocoaPods integration" error for plugin example projects

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#151504]
I just built `packages/go_router/example` and I see:
```
$ flutter build ios
All plugins found for ios are Swift Packages, but your project still has CocoaPods integration. Your project uses a non-standard Podfile and will need to be migrated to Swift Package Manager manually. Some steps you may need to complete include:
  * In the ios/ directory run "pod deintegrate"
  * Transition any Pod dependencies to Swift Package equivalents. See https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app
  * Transition any custom logic
  * Remove the include to "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig" in your ios/Flutter/Debug.xcconfig
  * Remove the include to "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig" in your ios/Flutter/Release.xcconfig

Removing CocoaPods integration will improve the project's build time.
Building com.example.example for device (ios-release)...
```

However, we want plugin authors to continue to support both CocoaPods and SwiftPM for the foreseeable future.  Can we detect the example project is in a plugin directory, and suppress the warning in that case?

/cc @loic-sharma

**Comment by stuartmorgan** [comment]
I thought this only showed up if the example was actually using SPM for everything. In that case the example isn't testing CocoaPods support anyway, and so they may as well `pod deintegrate`, right?

Actually continuing to actively test CocoaPods will require plugin authors turning off SPM support for some of their builds, IIUC.

**Comment by loic-sharma** [comment]
> I thought this only showed up if the example was actually using SPM for everything. 

That's correct. If plugin `foo` supports both SwiftPM & CocoaPods and the user enables SwiftPM, building `foo`'s example app will output this message.

> Actually continuing to actively test CocoaPods will require plugin authors turning off SPM support for some of their builds, IIUC.

Yup.

> In that case the example isn't testing CocoaPods support anyway, and so they may as well pod deintegrate, right?

Flutter plugin developers should test their plugins using _both_ SwiftPM and CocoaPods until further notice.

Flutter app developers can't use SwiftPM until they upgrade to the future Flutter SDK that enables SwiftPM. Adoption of the latest Flutter SDK version tends to be slow, so there'll be a transitionary period. Flutter plugin developers should fully support CocoaPods until most Flutter app developers are on a Flutter SDK that supports SwiftPM.

> Can we detect the example project is in a plugin directory, and suppress the warning in that case?

I like this idea! The warning incentivizes Flutter plugin developers to not test CocoaPods, which we don't want.

**Comment by stuartmorgan** [comment]
Oh, I was thinking about people seeing this because they have converted their plugin example, in which case the warning is correct. But I guess the issue is that if the example is auto-converted on the fly (e.g., in their CI), they will also see this warning, even though the plugin still supported CocoaPods as checked in.

Given that, I'm on board with suppressing the warning for now, but we should file an issue to track turning it back on in a couple of releases, since we don't want everyone example apps stuck depending on CocoaPods indefinitely either.

(FWIW, while we obviously would *like* people to, and I agree with not discouraging it with the tooling, I would expect that almost nobody actually *will* continue testing their plugin with CocoaPods once they convert, and will instead probably just rely on users filing issue if things break.)

**Comment by ryanheise** [comment]
> I'm on board with suppressing the warning for now

If it's actually not doing the conversion as intended, isn't that an error rather than a warning, and why would we suppress the message that informs the user that the intended conversion failed? I would like to hopefully add/remove `disable-swift-package-manager: true` to go back and forth, testing my plugin with and without SwiftPM.

As for the meaning of "a non-standard Podfile", I find the wording of this message confusing. Is it supposed to mean that I need to fix my Podfile into a more standard Podfile first in order for the tool to agree to do the conversion? That's my straightforward reading of the message, although my Podfile looks pretty normal/standard. Or, is the message supposed to mean that actually "any" Podfile is no longer considered to be standard? In which case, shouldn't it just say "a Podfile" to be absolutely clear that all Podfiles will error out, not just this subset of "non-standard" Podfiles?.

On a more practical note, is there any way I can convince the flutter tool to automatically convert my plugin example to SwiftPM when I get this message? This is for https://github.com/ryanheise/just_audio/pull/1394 and I'm currently getting this error when commenting out `disable-swift-package-manager: true` in the example's pubspec.yaml.

**Comment by ryanheise** [comment]
> Nothing _failed_ in this scenario, `flutter` just can't confidently remove all traces of CocoaPods from the project.

Ah, if that's what the message means, then I'm fine with ignoring it (and suppressing it in this case).

> In practice, it means that the Podfile does not exactly match what would be auto-created by the tool if the file were missing.

Then it's just as well that my Podfile is non-standard I suppose, otherwise the Flutter tool might have deleted all traces of it, and I don't want that!

Although it's a bit of a mystery why I didn't get the same message for the example project of audio_session, even though it has an identical Podfile to the problematic one in the just_audio example project.

**Comment by stuartmorgan** [comment]
> Although it's a bit of a mystery why I didn't get the same message for the example project of audio_session

At a glance, probably because it depends on `just_audio: ^0.9.42`. Unless you manually overrode that with a local version in your test, then you had at least one plugin in the build using CocoaPods (since SPM support for `just_audio` isn't published AFAICT).
